hypervisor/src/mshv/mod.rs

```diff
@@ -333,12 +303,27 @@ impl hypervisor::Hypervisor for MshvHypervisor {
                 msrs[pos].index = *index;
             }
 
+            #[cfg(feature = "sev_snp")]
+            let mem_size = _mem_size.unwrap_or_default();
+
+            #[cfg(feature = "sev_snp")]
+            let mem_size_for_bitmap = if mem_size as usize > 3 * ONE_GB {
+                mem_size as usize + ONE_GB
+            } else {
+                mem_size as usize
+            };
+
```

Comment:
I think this really requires a justification that why are we adding that extra 1GB in case guest memory is greater than 3G. Also isn't there a better way to get this information about PCI hole instead of hardcoding it as 1GB?

Reply:
Isn't the constant in arch/x86/layout.rs?

Reply:
This looks like this is the end of ram rather than the amount of memory (we use that nomenclature) elsewhere so it would be better to be consistent with that (see `end_of_ram_area` in memory_manager.rs - this will correctly handle holes, etc.

Reply:
@rbradford We create Vm object before Memory manager, How can I get that information during hypervisor vm creation?

Reply:
> @rbradford We create Vm object before Memory manager, How can I get that information during hypervisor vm creation?

You might need to refactor out the calculation - I think that value only depends on the configuration (and possibly some arch based constant values.)

Reply:
Or alternatively, and this would be my preference, you modify vm-memory `AtomicBitmap::enlarge()` and grow the size of the bitmap when an address is larger than the existing bitmap. All the changes would then be hidden inside the mshv module of the hypervisor crate (plus required vm-memory changes.)

Reply:
Cool. I am gonna make this PR as draft until I get the rust-vmm components updated

Reply:
You could look at using interior mutability with something like an `ArcSwap`.

Reply:
> You could look at using interior mutability with something like an `ArcSwap`.

@rbradford I see an issue using ArcSwap, ech vcpu object is cloning the bitmap from VM object. In any case if one vcpu object is enlarging the bitmap and replace the inner object inside ArcSwap does it propagate the changes the the bitmap that other vcpu object holds?

Reply:
In my recollection the changes does propagate, you may use `rcu` method provided by `arc_swap` to do that 🤔